core/HoistAuthModel.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2024 Extremely Heavy Industries Inc.
+ */
+import {HoistModel, PlainObject, XH} from './';
+
+/**
+ *  Base class for managing authentication lifecycle.
+ *
+ *  Hoist will consult this model early in  the initialization sequence, prior to
+ *  full Hoist initialization. This means that several core services (identity, configs, prefs)
+ *  will *not* be available, but it provides the app a hook to
+ *  do early service initialization or other work to support flows such as OAuth.
+ *
+ *  The base implementation of this class is minimal and would be adequate only for fully
+ *  transparent SSO based solutions such as NTLM.  Applications wishing to provide custom
+ *  authentication should provide an extended version of this class to `XH.renderApp`
+ */
+export class HoistAuthModel extends HoistModel {
+    /**
+     * Main Entry Point
+     *
+     * This method will be called *once* by Hoist early in the app loading lifecycle to determine
+     * if full app loading can proceed.
+     *
+     * It should return true if the user could be successfully authenticated, false if the user
+     * is not authenticated, and throw in the case of technical failure. If false is returned and
+     * `AppSpec.enableLoginForm` is true the framework will enter the 'loginRequired' state and
+     * show the user a *manual* login form.
```

Comment:
All logins are "manual", even the ones where I have to point the phone to my face for biometric auth.
Can we say "show the user a login form with two fields, username & password, which when submitted will be checked against...."  What will these credentials be checked against?  "...JESPA/NTLM or credentials set in the YML config for local development"?

Reply:
👍🏻